libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -539,48 +539,74 @@ internal class MapboxTripSession(
         tripService.updateNotification(progress)
         progress?.let {
             routeProgressObservers.forEach { it.onRouteProgressChanged(progress) }
-            checkBannerInstructionEvent(progress) { bannerInstruction ->
+            bannerInstructionEvent.isOccurring(progress)
+            checkBannerInstructionEvent { bannerInstruction ->
                 bannerInstructionsObservers.forEach {
                     it.onNewBannerInstructions(bannerInstruction)
                 }
             }
-            checkVoiceInstructionEvent(progress) { voiceInstruction ->
+            voiceInstructionEvent.isOccurring(progress)
+            checkVoiceInstructionEvent { voiceInstruction ->
                 voiceInstructionsObservers.forEach {
                     it.onNewVoiceInstructions(voiceInstruction)
                 }
             }
         }
     }
 
+    private fun checkLatestValidBannerInstructionEvent(
+        action: (BannerInstructions) -> Unit
+    ) {
+        ifNonNull(bannerInstructionEvent.latestBannerInstructions) {
+            action(it)
+        }
+    }
+
+    private fun invalidateLatestBannerInstructionEvent() {
+        if (bannerInstructionsObservers.isEmpty()) {
+            bannerInstructionEvent.invalidateLatestBannerInstructions()
```

Comment:
I think this should be called when the route changes, not when observers are unsubscribed, shouldn't it?